### PR TITLE
feat: support RAS default prompts wizard UI

### PR DESCRIPTION
**Pull request by dkoo**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Supports new RAS-related setup UI in https://github.com/Automattic/newspack-plugin/pull/2399. The main changes:

- Updates the prompt schema to support additional user input field data types (array and integer in addition to string)
- Implements logic to store user inputs by field name and populate preset config placeholders with matching stored inputs when fetching the preset configs
- Implements logic to build dynamic prompt previews from preset config + user input, without having to create prompt posts

### How to test the changes in this Pull Request:

Follow testing instructions in https://github.com/Automattic/newspack-plugin/pull/2399. The plugin should function as usual and all tests should continue passing even without the changes in https://github.com/Automattic/newspack-plugin/pull/2399.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: includes/class-newspack-popups-model.php [hunk 1]
```diff
@@ -240,31 +478,108 @@ public static function retrieve_preview_popup( $post_id ) {
...
+			'trigger_type'                   => filter_input( INPUT_GET, Newspack_Popups::PREVIEW_QUERY_KEYS['trigger_type'], FILTER_SANITIZE_FULL_SPECIAL_CHARS ),
+			'trigger_delay'                  => filter_input( INPUT_GET, Newspack_Popups::PREVIEW_QUERY_KEYS['trigger_delay'], FILTER_SANITIZE_FULL_SPECIAL_CHARS ),
+			'trigger_scroll_progress'        => filter_input( INPUT_GET, Newspack_Popups::PREVIEW_QUERY_KEYS['trigger_scroll_progress'], FILTER_SANITIZE_FULL_SPECIAL_CHARS ),
+			'trigger_blocks_count'           => filter_input( INPUT_GET, Newspack_Popups::PREVIEW_QUERY_KEYS['trigger_blocks_count'], FILTER_SANITIZE_FULL_SPECIAL_CHARS ),
+			'archive_insertion_posts_count'  => filter_input( INPUT_GET, Newspack_Popups::PREVIEW_QUERY_KEYS['archive_insertion_posts_count'], FILTER_SANITIZE_FULL_SPECIAL_CHARS ),
+			'archive_insertion_is_repeating' => filter_input( INPUT_GET, Newspack_Popups::PREVIEW_QUERY_KEYS['archive_insertion_is_repeating'], FILTER_VALIDATE_BOOLEAN ),
+			'utm_suppression'                => filter_input( INPUT_GET, Newspack_Popups::PREVIEW_QUERY_KEYS['utm_suppression'], FILTER_SANITIZE_FULL_SPECIAL_CHARS ),
+		];
```
Comment (miguelpeixe) [inline includes/class-newspack-popups-model.php:570]: NIT: We can avoid code repetition by looping through a map to build this:

```php
$options_filters = [
	'background_color' => FILTER_SANITIZE_FULL_SPECIAL_CHARS
	'display_type' => FILTER_VALIDATE_BOOLEAN
];
$options = [];
foreach( $options_filters as $option => $filter ) {
	$options[ $option ] = filter_input( INPUT_GET, Newspack_Popups::PREVIEW_QUERY_KEYS[ $option ], $filter );
}
```

File: includes/class-newspack-popups-model.php [hunk 2]
```diff
@@ -240,31 +478,108 @@ public static function retrieve_preview_popup( $post_id ) {
...
+				'trigger_type'                   => filter_input( INPUT_GET, Newspack_Popups::PREVIEW_QUERY_KEYS['trigger_type'], FILTER_SANITIZE_FULL_SPECIAL_CHARS ),
+				'trigger_delay'                  => filter_input( INPUT_GET, Newspack_Popups::PREVIEW_QUERY_KEYS['trigger_delay'], FILTER_SANITIZE_FULL_SPECIAL_CHARS ),
+				'trigger_scroll_progress'        => filter_input( INPUT_GET, Newspack_Popups::PREVIEW_QUERY_KEYS['trigger_scroll_progress'], FILTER_SANITIZE_FULL_SPECIAL_CHARS ),
+				'trigger_blocks_count'           => filter_input( INPUT_GET, Newspack_Popups::PREVIEW_QUERY_KEYS['trigger_blocks_count'], FILTER_SANITIZE_FULL_SPECIAL_CHARS ),
+				'archive_insertion_posts_count'  => filter_input( INPUT_GET, Newspack_Popups::PREVIEW_QUERY_KEYS['archive_insertion_posts_count'], FILTER_SANITIZE_FULL_SPECIAL_CHARS ),
+				'archive_insertion_is_repeating' => filter_input( INPUT_GET, Newspack_Popups::PREVIEW_QUERY_KEYS['archive_insertion_is_repeating'], FILTER_VALIDATE_BOOLEAN ),
+				'utm_suppression'                => filter_input( INPUT_GET, Newspack_Popups::PREVIEW_QUERY_KEYS['utm_suppression'], FILTER_SANITIZE_FULL_SPECIAL_CHARS ),
```
Comment (miguelpeixe) [inline includes/class-newspack-popups-model.php:501]: Same as per the previous comment, could improve maintainability by reducing code repetition. Maybe a separate method that handles both cases?

**Review by miguelpeixe — APPROVED** [review]
Working as described!

Just a couple of non-blocking comments:

File: includes/class-newspack-popups-model.php (hunk 1 above)
Comment (dkoo) [reply includes/class-newspack-popups-model.php]: Good idea—implemented in 49cdc86.

File: includes/class-newspack-popups-model.php (hunk 2 above)
Comment (dkoo) [reply includes/class-newspack-popups-model.php]: Good idea—implemented in 49cdc86.

**Review by miguelpeixe — APPROVED** [review]
🙌

**Pull request merged**